src/plugins/intel_cpu/src/cpu_memory.cpp

```diff
@@ -31,19 +32,26 @@ BlockedMemoryDescPtr IMemory::getDescWithType<BlockedMemoryDesc, 0, 0>() const {
 }
 
 namespace {
-inline void setSubnormalsToZero(float* data, size_t size) {
+inline void setSubnormalsToZeroAndbf16Saturation(float* data, size_t size, bool ftz, bool bf16saturation) {
     uint32_t* u32data = reinterpret_cast<uint32_t*>(data);
+    float* floatdata = reinterpret_cast<float*>(data);
     for (size_t i = 0; i < size; ++i) {
-        if ((u32data[i] & (0xFF << 23)) == 0) {
+        if (ftz && ((u32data[i] & (0xFF << 23)) == 0)) {
```

Comment:
Have you checked that the compiler extracts ftz out of the loop? I would like to make sure we won't drop subnormals elimination performance as it might be applied for quite a big tensors.

Reply:
To make sure ftz and bf16saturation checked outside of the loop. these two variables are moved outside manually.